connect/runtime/src/main/java/org/apache/kafka/connect/runtime/distributed/DistributedHerder.java

```diff
@@ -573,9 +582,11 @@ public Void call() throws Exception {
                     return null;
                 }
 
-                if (worker.ownsConnector(connName)) {
+                if (assignment.connectors().contains(connName)) {
```

Comment:
Hmm, interesting. I think this kind of makes sense, but is also a bit weird if you run multiple connectors per process. I guess at that point you're just accepting that your tasks won't be isolated?